src/bp-forums/groups.php

```diff
@@ -317,6 +632,93 @@ public function edit_screen( $group = false ) {
 			}
 		}
 
+		/**
+		 * Set the selected forum in the forum dropdown from the group edit screen.
+		 *
+		 * @since BuddyBoss 1.5.7
+		 *
+		 * @param string $where        where condition forum post query.
+		 * @param object $query_object forum post query property.
+		 * 
+		 * @return string
+		 */
+		public function update_forum_dropdown_qeury( $where, $query_object ) {
+			global $wpdb;
+
+			// Prevent update query when post type is not forum.
+			if ( $query_object->query['post_type'] != 'forum' ) {
+				return $where;
+			}
+
+			$forum_ids = bbp_get_group_forum_ids( $this->group_id );
+			$forum_ids = array_filter( $forum_ids );
+			
+			if ( empty( $forum_ids ) ) {
+				return $where;
+			}
+
+			$query_string = implode( ',', $forum_ids );
+			
+			// Set the selected forum in the forum dropdown from the group edit screen.
+			$where .= " OR {$wpdb->postmeta}.post_id IN ($query_string)";
+
+			return $where;
+		}
+
+		/**
+		 * Exclude the forums
+		 *
+		 * Exclude if the forum associated with other groups.
+		 * Exclude if the forum type is category.
+		 * Exclude if the forum is child forum.
+		 * 
+		 *
+		 * @since BuddyBoss 1.5.7
+		 *
+		 * @param array $forum_ids
+		 * @param int   $group_id
+		 * @uses bbp_get_forum() To get the forum.
+		 *
+		 * @return array
+		 */
+		public function get_validate_forum_ids( $forum_ids, $group_id ) {
```

Comment:
change the function name to `validate_forum_ids`

Reply:
Update https://github.com/buddyboss/buddyboss-platform/pull/2117/commits/414560ca0a1366f661b41fee22788390ea057d55